app/models/post.rb

```diff
@@ -1,2 +1,18 @@
 class Post < ApplicationRecord
+  belongs_to :user
+
+  after_save :update_comment_counter
+
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
```

Comment:
- [**Optional**] Great job for making use of [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method. But it seems you forgot to make the `update_post_counter` method private. Kindly check the link once more